docs/TrafficSplitting.md

```diff
@@ -0,0 +1,157 @@
+
+`kn` client can work with following properties of a traffic target defined in traffic block of a service spec.
+ - **RevisionName**: RevisionName of a specific revision to which to send a portion of traffic.
+ - **Tag**: Tag is optionally used to expose a dedicated url for referencing this target exclusively.
+ - **Percent**: Percent specifies percent of the traffic to this Revision.
+
+## Operations
+
+Operations required to perform traffic splitting use cases, for example: blue/green deployment, etc can be summarized as below:
+ - Tag targets
+ - Split traffic
+
+### Tag Targets
+
+User can tag traffic targets (revisions) to
+ - Create custom URLs
+ - Create a human-friendly reference to refer to a traffic target
+ - Create static routes and change the underlying revisions as needed
+ - Remove the custom URLs OR remove the tags of targets
+ - Remove a target from traffic block altogether
+
+
+#### Tagging commands
+
+#### 1. `kn service tag-target serviceName [RevisionName1:Tag1] [RevisionName2:Tag2]`
+
+ - Takes the first argument as service name to operate on
+ - Tags given revision with specified tag thereby creating a custom URL
+ - Does not change the traffic percent of the target
+ - Multiple `RevisionName:Tag` combination can be specified
+ - Multiple tags for same revision can be specified
+   - `kn service tag-target svc echo-v1:stable echo-v1:current`
+ - If the target doesn't exist in deployed service, it creates a target entry with 0% traffic portion
+ - For updating tags of existing traffic targets, use `kn service untag-target` followed by `kn service tag-target`, because
+    - The deployed service might have multiple tags referring to a `RevisionName`
+    - The specified `RevisionName:Tag` combination on CLI might request to have different tags for same revision
+
+#### 2. `kn service untag-target serviceName [Tag] [RevisionName:Tag] [RevisionName [--all]]`
+
+ - Takes the first argument as service name to operate on
+ - Takes single reference of a traffic target
+ - Does not change the traffic percent of the target
+ - Can be specified either via
+   -  `Tag`, removes the specified tag from traffic targets
+   -  `RevisionName`, removes the tag for target where `RevisionName` matches, errors if multiple tags for a revision exists
+   -  `RevisionName:Tag` removes `Tag` for `RevisionName`, useful for case where a single revision is tagged with multiple tags
+   - `--all` flag can be optionally specified along with `RevisionName` to remove all tags for given revision
+ - If a target has 0% traffic portion and it is untagged, target is removed from traffic block altogether
+
+
+## Split Traffic
+
+A user can
+ - Specify the traffic percent portion to route to a particular target
+ - Adjust the traffic percent of existing traffic targets
+ - Reference a traffic target either by `RevisionName` or `Tag`
```

Comment:
We should mention that in the future people may be able to specify something other than "percent" as the criteria, so we need to design a syntax that doesn't preclude us from supporting that later on.

Since right now I believe all of our proposals have done something of the form `...:%` - then as long as the `%` specification is at the end of the syntax then I think we might be ok. We can then assume that whatever string is after the `:` is the criteria - even if it's a complex expression. We may need to allow for it to be of the form `type:expression` (e.g. `path:/foo` or `header:user=john`) but I think that's workable.  So, nothing needs to be changed w.r.t. any proposal (except the comment above to mention it) - I just wanted to bring this up so people are thinking about it.